sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/dynamodb/DynamoDBIOReadTest.java

```diff
@@ -75,6 +76,29 @@ public void testReadOneSegment() {
     pipeline.run().waitUntilFinish();
   }
 
+  @Test
+  public void testReadWithCustomLimit() {
```

Comment:
The only remaining test case from `DynamoDBIOTest` that wasn't already covered in `DynamoDBIOReadTest` or `DynamoDBIOWriteTest`